Doc for deviated PoroElastic wellbore (#1585)
GEOSX/GEOSX: src/docs/sphinx/advancedExamples/validationStudies/deviatedPoroElasticWellbore/Example.rst

.. _AdvancedExampleDeviatedPoroElasticWellbore:

####################################################
Deviated Poro-Elastic Wellbore Problem
####################################################

------------------------------------------------------------------
Problem description
------------------------------------------------------------------

This example aims to solve a typical injection problem of a deviated poroelastic open wellbore subjected to a fluid pressure injected inside the hole. The problem geometry is generated by the interal wellbore mesh generator. The poroelastic coupled solver, that is a combination of the solid mechanic solver and the single phase flow solver, is employed to solver this coupling problem. In-situ stress and pressure are excluded from this example.

> Analytical results of the fluid pressure, and the radial and hoop stresses around the wellbore are expressed in the Laplace space as `(Detournay and Cheng, 1988) <https://www.sciencedirect.com/science/article/abs/pii/0148906288922991>`__ :

Reviewer: ```suggestion
Analytical solutions of the pore pressure, the radial and hoop stresses in the near wellbore region are expressed in the Laplace space as `(Detournay and Cheng, 1988) <https://www.sciencedirect.com/science/article/abs/pii/0148906288922991>`__ :
```

Author: Done

.. math::
    p = p_0 \frac{kn(0,R \sqrt{s})}{s kn(0,\sqrt{s})}

.. math::
    \sigma_{rr} = -b \frac{1-2\nu}{1-\nu} p_0 \frac{-R * kn(1,R \sqrt{s}) + kn(1,\sqrt{s})}{R^2 \sqrt{s^3} kn(0,\sqrt{s})}

.. math::
    \sigma_{\theta\theta} = -b \frac{1-2\nu}{1-\nu} p - \sigma_{rr}

> where :math:`s` is the Laplace variable, :math:`R` is the dimensionless radial coordinate that is defined by the radial coordinate normalized by the hole radius, :math:`\nu` the Poisson ratio and :math:`b` the Biot coefficient. Fluid pressure and stresses in time space are obtained from these analytical expression by the inverse Laplace transformation (see the attached Python script for more details).  

Reviewer: ```suggestion
where :math:`s` is the Laplace variable, :math:`kn` is the is the modified Bessel function of second kind of order n, :math:`R` is the dimensionless radial coordinate that is defined by the radial coordinate normalized by the wellbore radius, :math:`\nu` is the Poisson ratio and :math:`b` is the Biot coefficient. Fluid pressure and stresses in time space are obtained from these analytical expressions by inversing Laplace transformation (see the attached Python script for more details).  
```

Reviewer: Please verify the analytical solution. Here, `s` should be an expression of Laplace variable and diffusion coefficient. This analytical solutions are expressed in the local coordinate system (along wellbore trajectory). I suggest to make a note here.

Author: Done

**Input file**

Everything required is
contained within a single GEOSX input file that is located at:

.. code-block:: console

  src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml

In this example, we would focus our attention on the ``Solver`` and the ``Mesh`` tags.

-----------------------------------------------------------
Poroelastic solver
-----------------------------------------------------------

> The coupled ``Poroelastic`` solver, that is a coupling between the solid mechanic solver ``SolidMechanicsLagrangianSSLE`` and the single phase flow solver ``SinglePhaseFVM``, is required for solving this wellbore problem.  

Reviewer: ```suggestion
The coupled ``Poroelastic`` solver, that defines a coupling strategy between the solid mechanics solver ``SolidMechanicsLagrangianSSLE`` and the single phase flow solver ``SinglePhaseFVM``, is required for solving this wellbore problem.  
```

Author: Done

 
.. literalinclude:: ../../../../../../src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml
  :language: xml
  :start-after: <!-- SPHINX_WellboreMesh -->
  :end-before: <!-- SPHINX_WellboreMeshEnd -->

--------------------------------------------------------------------   
Constitutive law
--------------------------------------------------------------------

> Isotropic ``PoroElasticIsotropic`` constitutive behavior, with the specified bulk and shear elastic moduli and the Biot coefficient, is considered for the rock around the wellbore. Fluid constitutive properties such as dynamic viscosity, compressibility, etc. are given in the ``CompressibleSinglePhaseFluid`` constitutive block. 

Reviewer: ```suggestion
Isotropic constitutive behavior ``PoroElasticIsotropic`` , with the specified bulk and shear elastic moduli, is considered for the rock around the wellbore. Fluid properties, such as dynamic viscosity and compressibility, are given in the ``CompressibleSinglePhaseFluid`` constitutive block. 
```

Author: Done

.. literalinclude:: ../../../../../../src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml
  :language: xml
  :start-after: <!-- SPHINX_Material -->
  :end-before: <!-- SPHINX_MaterialEnd -->

--------------------------------------------------------------------	       
Boundary conditions
--------------------------------------------------------------------

> Far-field boundaries are subjected to roller constraints and impermeable condition. The pressure on the wellbore wall is defined by face pressure field specification. The nodeset generated by the internal wellbore generator for this face is named as ``rneg``. The negative sign of the scale is attributed to the negative sign convention for compressive stresses in GEOSX. Initial fluid pressure, reference porosity and permeability are also given for the whole domain. In this example, uniform isotropic permeability is assumed.

Reviewer: ```suggestion
Far-field boundaries are impermeable and subjected to roller constraints. The pressure on the wellbore wall is defined by face pressure field specification. The nodeset generated by the internal wellbore generator for this face is named as ``rneg``. The negative sign of the scale denotes the fluid injection. Initial fluid pressure, reference porosity and permeability are also given for the computational domain. In this example, uniform isotropic permeability is assumed.
```

Author: Done

.. literalinclude:: ../../../../../../src/coreComponents/physicsSolvers/multiphysics/benchmarks/DeviatedPoroElasticWellbore_Injection.xml
  :language: xml
  :start-after: <!-- SPHINX_BoundaryConditions -->
  :end-before: <!-- SPHINX_BoundaryConditionsEnd -->

---------------------------------
Results and benchmark
---------------------------------

> Result of the fluid pressure distribution at an injection time of 78 (s) is shown in the figure below:

Reviewer: ```suggestion
Result of the fluid pressure distribution after 78 s injection is shown in the figure below:
```

Author: Done
